### PR TITLE
feat: migrate customdb to new Atlas SDK

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ [INTMDB-1086](https://jira.mongodb.org/browse/INTMDB-1086)

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Manual QA performed:

- [x] cfn invoke for each of CRUDL/cfn test
- [x] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [x] Published to AWS private registry
- [x] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [x] Deleted stack to ensure resources are deleted
- [x] Created multiple resources in same stack
- [x] Validated in Atlas UI
- [x] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

#### CFN test
```
cfn test --function-name TestEntrypoint --verbose

13 passed, 2 skipped, 9 deselected in 68.19s (0:01:08)
```

#### CFN stack
Screenshot 2023-09-27 at 14 31 31
Screenshot 2023-09-27 at 14 31 15

### CODE REVIEW

File: cfn-resources/custom-db-role/cmd/resource/resource.go [hunk 1]
```diff
@@ -49,31 +47,25 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
...
 		return *peErr, nil
 	}
 
 	atlasCustomDBRole := currentModel.ToCustomDBRole()
-
-	customDBRole, response, err := client.CustomDBRoles.Create(context.Background(), *currentModel.ProjectId, &atlasCustomDBRole)
+	customDBRole, response, err := client.AtlasV2.CustomDatabaseRolesApi.CreateCustomDatabaseRole(context.Background(), *currentModel.ProjectId, atlasCustomDBRole).Execute()
 	if err != nil {
-		if response.Response.StatusCode == http.StatusConflict {
+		if response.StatusCode == http.StatusConflict {
```
Comment (gssbzn) [inline cfn-resources/custom-db-role/cmd/resource/resource.go:59]: FYI `apiError, ok := admin.AsError(err)`
```suggestion
		if apiError.Error == http.StatusConflict {
```
and you can ignore the raw http response

File: cfn-resources/custom-db-role/test/cfn-test-create-inputs.sh [hunk 2]
```diff
@@ -11,7 +11,7 @@ set -o pipefail
 set -x
 
 function usage {
-    echo "Creates a new customdb role for the test"
+	echo "Creates a new customdb role for the test"
```
Comment (gssbzn) [inline cfn-resources/custom-db-role/test/cfn-test-create-inputs.sh:14]: `shfmt`?

**Review by gssbzn — APPROVED** [review]
LGTM from an SDK/GO PoV

File: cfn-resources/custom-db-role/test/cfn-test-create-inputs.sh (hunk 2 above)
Comment (andreaangiolillo) [reply cfn-resources/custom-db-role/test/cfn-test-create-inputs.sh:14]: we do use `shfmt` as part of the pre-commit hook https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/57f6ace81eb39e80ac7ffbf86deee47da84b5509/.githooks/pre-commit#L26

I think this is the reason why I add these changes here

**Pull request merged**